webhook-update.js

```diff
@@ -0,0 +1,53 @@
+
+/* update existing hooks */
+const doAsync = require("doasync");
+var GH = require("./gh");
+var config = require("./config.json");
+var hookURL = config.hookURL;
+
+
+if (require.main === module) {
+    var Store = require("./store");
+    config.logToConsole = false;
+    const store = new Store(config);
+    store.repos(async (err, data) => {
+        for (repo of data) {
+            const owner = repo.owner;
+            const shortname = repo.name;
+            const { token } = await doAsync(store).getToken(owner);        
+            const gh = new GH({ accessToken: token });
+            try {
+                const { data: hooks } = await gh.octo.request("GET /repos/:owner/:name/hooks",
+                {
+                    owner: owner,
+                    name: shortname
+                });
+
+                const hook = (hooks || hooks.length) ? hooks.find(function(h) { return h && h.config && h.config.url === hookURL; }) : null;
+
+                if (hook) {
+                    const secret = await doAsync(store).getSecret(`${owner}/${shortname}`);
+                    await gh.octo.request("PATCH /repos/:owner/:name/hooks/:hook",
```

Comment:
this probably needs a `try` / `catch`

Reply:
addressed in https://github.com/w3c/ash-nazg/pull/224/commits/2d70617d87ca1aa4f0b68f01bbab15039ddcbd2a